app/controllers/application_controller/tenancy.rb

```diff
@@ -9,11 +9,10 @@ module ApplicationController::Tenancy
   end
 
   def current_tenant
-    # current_user.try(:current_group).try(:tenant) || Tenant.default_tenant
     @current_tenant ||=
```

Comment:
1. We call this a number of times. we want some form of caching
2. if they change group/tenant, then they will come in on a new request.
3. this is cached for just the life of this http request.
4. logging out/ logging in - different request (note, `@current_user` is also cached)